config/management/src/lib.rs

```diff
@@ -287,16 +271,61 @@ impl Command {
 
         writeln!(buffer, "{} - {}", key, value).unwrap();
     }
+
+    fn submit_key(key_name: &str, secure_backends: SecureBackends) -> Result<Ed25519PublicKey, Error> {
+        let local: Box<dyn Storage> = secure_backends.local.try_into()?;
+        if !local.available() {
+            return Err(Error::LocalStorageUnavailable);
+        }
+
+        let key = local
+            .get_public_key(key_name)
+            .map_err(|e| Error::LocalStorageReadError(e.to_string()))?
+            .public_key;
+
+        if let Some(remote) = secure_backends.remote {
+            let key = Value::Ed25519PublicKey(key.clone());
+            let mut remote: Box<dyn Storage> = remote.try_into()?;
+            if !remote.available() {
+                return Err(Error::RemoteStorageUnavailable);
+            }
+
+            remote
+                .create_with_default_policy(key_name, key)
+                .map_err(|e| Error::LocalStorageReadError(e.to_string()))?;
+        }
+
+        Ok(key)
+    }
 }
 
 #[derive(Debug, StructOpt)]
-pub struct SecureBackend {
-    #[structopt(long)]
-    host: String,
+pub struct SecureBackends {
+    /// The local secure backend. Secure backends are represented as a
+    /// semi-colon deliminted key value pair: "k0=v0;k1=v1;...".
+    /// The current supported formats are:
+    ///     Vault: "backend=vault;server=URL;token=TOKEN"
+    ///         vault has an optional namespace: "namespace=NAMESPACE"
+    ///     InMemory: "backend=memory"
+    ///     OnDisk: "backend=disk;path=LOCAL_PATH"
+    #[structopt(long, verbatim_doc_comment)]
+    local: SecureBackend,
+    /// The remote secure backend. See the comments for the local backend.
     #[structopt(long)]
-    token: String,
-    #[structopt(long)]
-    namespace: Option<String>,
+    remote: Option<SecureBackend>,
+}
+
+#[derive(Debug, StructOpt)]
+pub struct SingleBackend {
+    /// The secure backend. Secure backends are represented as a semi-colon
```

Comment:
You don't think it's maybe clearer to have this comment above the "SecureBackend" struct, rather than having to duplicate it and place it above all users of the struct?

Reply:
this is the problem of structopt :/

I imagine there might be some way by looking at how clap works (the library structopt is build on top of)